txp/Pos/Txp/Logic/Global.hs

```diff
@@ -51,13 +52,12 @@ verifyBlocks
        TxpGlobalVerifyMode m
     => Bool -> OldestFirst NE TxpBlock -> m (OldestFirst NE TxpUndo)
 verifyBlocks verifyAllIsKnown newChain = do
-    let epoch = NE.last (getOldestFirst newChain) ^. choosing epochIndexL (_1 . epochIndexL)
+    let epoch = NE.last (getOldestFirst newChain) ^. epochIndexL
     fst <$> runToilAction @_ @() (mapM (verifyDo epoch) newChain)
   where
     verifyDo epoch = verifyToil epoch verifyAllIsKnown . convertPayload
     convertPayload :: TxpBlock -> [TxAux]
-    convertPayload (Left _)             = []
-    convertPayload (Right (_, payload)) = flattenTxPayload payload
+    convertPayload payload = flattenTxPayload (bcmPayload payload)
```

Comment:
How does it work if `payload` is `ComponentBlockGenesis`? Will `bcmPayload` fail? You should pattern match here I suppose.